sensor/common/deploymentenhancer/component.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/stackrox/rox/sensor/common/store"
+)
+
+var (
+	log                 = logging.LoggerForModule()
+	deploymentQueueSize = 50 // TODO(ROX-21291): Configurable via env var
+)
+
+// The DeploymentEnhancer takes a list of Deployments and enhances them with all available information
+type DeploymentEnhancer struct {
+	responsesC       chan *message.ExpiringMessage
+	deploymentsQueue chan *central.DeploymentEnhancementRequest
+	storeProvider    store.Provider
+}
+
+// CreateEnhancer creates a new Enhancer
+func CreateEnhancer(provider store.Provider) common.SensorComponent {
+	return &DeploymentEnhancer{
+		responsesC:       make(chan *message.ExpiringMessage),
+		deploymentsQueue: make(chan *central.DeploymentEnhancementRequest, deploymentQueueSize),
+		storeProvider:    provider,
+	}
+}
+
+// ProcessMessage takes an incoming message and queues it for enhancement
+func (d *DeploymentEnhancer) ProcessMessage(msg *central.MsgToSensor) error {
+	toEnhance := msg.GetDeploymentEnhancementRequest()
+	if toEnhance == nil {
+		return nil
+	}
+	log.Debugf("Received message to process in DeploymentEnhancer: %v++", msg)
+	d.deploymentsQueue <- toEnhance
+	return nil
+}
+
+// Start starts the component
+func (d *DeploymentEnhancer) Start() error {
+	go func() {
+		for {
+			deploymentMsg, more := <-d.deploymentsQueue
+			if !more {
+				return
+			}
+			requestID := deploymentMsg.GetMsg().GetId()
+			if requestID == "" {
+				log.Warnf("received deploymentEnhancement msg with empty request ID. Discarding request.")
+				continue
+			}
+			deployments := d.extractAndEnrichDeployments(deploymentMsg)
+			d.sendDeploymentsToCentral(requestID, deployments)
+		}
+	}()
+	return nil
+}
+
+func (d *DeploymentEnhancer) extractAndEnrichDeployments(deploymentMsg *central.DeploymentEnhancementRequest) []*storage.Deployment {
+	var ret []*storage.Deployment
+
+	deployments := deploymentMsg.GetMsg().GetDeployments()
+	if deployments == nil {
+		log.Warnf("received deploymentEnhancement msg with no deployments")
+		return ret
+	}
+
+	log.Debugf("Received deploymentEnhancement msg with %v deployment(s)", len(deploymentMsg.GetMsg().GetDeployments()))
+	for _, deployment := range deployments {
+		enriched, err := d.enrichDeployment(deployment)
+		if err != nil {
+			log.Warnf("Failed to enrich deployment: %v", deployment)
+			continue
+		}
+		ret = append(ret, enriched)
+	}
+
+	return ret
+}
+
+func (d *DeploymentEnhancer) sendDeploymentsToCentral(id string, deployments []*storage.Deployment) {
+	d.responsesC <- message.New(&central.MsgFromSensor{
+		Msg: &central.MsgFromSensor_DeploymentEnhancementResponse{
+			DeploymentEnhancementResponse: &central.DeploymentEnhancementResponse{
+				Msg: &central.DeploymentEnhancementMessage{
+					Id:          id,
+					Deployments: deployments,
+				},
+			},
+		},
+	})
+}
+
+func (d *DeploymentEnhancer) enrichDeployment(deployment *storage.Deployment) (*storage.Deployment, error) {
+	localImages := d.findLocalImages(deployment)
```

Comment:
Do we really need local images? It's not being used in `EnhanceDeploymentReadOnly`.